Move CreateRWLayer() parameters in a struct

rhvgoyal opened the pull request:
Move some of the optional parameters of CreateRWLayer() in a struct
called CreateRWLayerOpts. This will make it easy to add more options
arguments without having to change signature of CreateRWLayer().

Signed-off-by: Vivek Goyal <vgoyal@redhat.com>

rhvgoyal commented:
cc @dmcgowan

dmcgowan commented on layer/layer_store.go:
```diff
@@ -444,7 +444,17 @@ func (ls *layerStore) Release(l Layer) ([]Metadata, error) {
 
-func (ls *layerStore) CreateRWLayer(name string, parent ChainID, mountLabel string, initFunc MountInit, storageOpt map[string]string) (RWLayer, error) {
+func (ls *layerStore) CreateRWLayer(name string, parent ChainID, opts *CreateRWLayerOpts) (RWLayer, error) {
+	var storageOpt map[string]string
```
nit: put these in `var(...)`

rhvgoyal replied:
will do.

rhvgoyal committed:
Move CreateRWLayer() parameters in a struct

Move some of the optional parameters of CreateRWLayer() in a struct
called CreateRWLayerOpts. This will make it easy to add more options
arguments without having to change signature of CreateRWLayer().

Signed-off-by: Vivek Goyal <vgoyal@redhat.com>

rhvgoyal commented:
Moved variables inside var() block.

dmcgowan commented:
LGTM

runcom commented:
LGTM

thaJeztah commented:
all green, merging
